### PR TITLE
Fix PortChannel GET at higher level container

### PR DESCRIPTION
**Pull request by Tejaswi-Goel**
**Before fix only partial config and no state data returned :** 
curl -H "accept: application/yang-data+json" -k -u admin:broadcom -X GET "https://10.11.133.22/restconf/data/openconfig-interfaces:interfaces/interface=PortChannel60/openconfig-if-aggregate:aggregation"
{
"openconfig-if-aggregate:aggregation": {
"config":
{ "lag-type": "LACP" }
}
}

**After fix complete data returned:**
root@sonic:/tmp# curl -kX GET "https://10.11.203.17/restconf/data/openconfig-interfaces:interfaces/interface=PortChannel200/openconfig-if-aggregate:aggregation" 
{"openconfig-if-aggregate:aggregation":{"**config**":{"lag-type":"LACP","min-links":2,"openconfig-interfaces-ext:fallback":true},"openconfig-vlan:**switched-vlan**":{"config":{"access-vlan":100},"state":{"access-vlan":100}},"**state**":{"lag-type":"LACP","member":["Ethernet20"],"min-links":2,"openconfig-interfaces-ext:fallback":true}}}

**Before fix following request returned empty data:** 
curl -kX GET "https://10.11.203.17/restconf/data/openconfig-interfaces:interfaces/interface=PortChannel100/openconfig-if-aggregate:aggregation/openconfig-vlan:switched-vlan" 
{}
**After fix:** 
root@sonic:/tmp# curl -kX GET "https://10.11.203.17/restconf/data/openconfig-interfaces:interfaces/interface=PortChannel200/openconfig-if-aggregate:aggregation/openconfig-vlan:switched-vlan" 
{"openconfig-vlan:switched-vlan":{"config":{"access-vlan":100},"state":{"access-vlan":100}}}r

Signed-off-by: Tejaswi Goel <Tejaswi_Goel@Dell.com>

### CODE REVIEW

File: src/translib/transformer/sw_portchannel.go [hunk 1]
```diff
@@ -167,6 +179,35 @@ var YangToDb_lag_min_links_xfmr FieldXfmrYangToDb = func(inParams XfmrParams) (m
...
+    result := make(map[string]interface{})
+
+    intfType, _, ierr := getIntfTypeByName(inParams.key)
+    if ierr != nil || intfType != IntfTypePortChannel  {
+        log.Error("Invalid Interface")
+        return result, err
+    }
+    data := (*inParams.dbDataMap)[inParams.curDb]
+    links, ok := data[PORTCHANNEL_TABLE][inParams.key].Field["min_links"]
+    if ok {
```
Comment (ArthiSivanantham) [inline src/translib/transformer/sw_portchannel.go:204]: Can't we use the existing get_min_links() API here ?

File: src/translib/transformer/sw_portchannel.go [hunk 2]
```diff
@@ -247,6 +298,32 @@ var YangToDb_lag_fallback_xfmr FieldXfmrYangToDb = func(inParams XfmrParams) (ma
...
+
+    intfType, _, ierr := getIntfTypeByName(inParams.key)
+    if ierr != nil || intfType != IntfTypePortChannel {
+        log.Error("Invalid Interface")
+        return result, err
+    }
+
+    data := (*inParams.dbDataMap)[inParams.curDb]
+
+    fallback, ok := data[PORTCHANNEL_TABLE][inParams.key].Field["fallback"]
```
Comment (ArthiSivanantham) [inline src/translib/transformer/sw_portchannel.go:324]: Could we use get_fallback() API ?

File: src/translib/transformer/sw_portchannel.go (hunk 1 above)
Comment (Tejaswi-Goel) [reply src/translib/transformer/sw_portchannel.go:204]: get_min_links() again makes a call to DB to get min-links, (*inParams.dbDataMap) will be more efficient to get min-links value. I have reduced some repetitive code by adding new function  uint16Conv

File: src/translib/transformer/sw_portchannel.go (hunk 2 above)
Comment (Tejaswi-Goel) [reply src/translib/transformer/sw_portchannel.go:324]: get_fallback() again makes a call to DB, using (*inParams.dbDataMap) will be more efficient.

**Comment by Tejaswi-Goel** [comment]
More testing after rebasing with latest code: 

curl -kX GET "https://10.11.203.17/restconf/data/openconfig-interfaces:interfaces/interface=PortChannel200/**openconfig-if-aggregate:aggregation/state"
{"openconfig-if-aggregate:state":{"lag-type":"LACP","member":["Ethernet20"],"min-links":2,"openconfig-interfaces-ext:fallback":true}}

 curl -kX GET "https://10.11.203.17/restconf/data/openconfig-interfaces:interfaces/interface=PortChannel200**/openconfig-if-aggregate:aggregation/config"
{"openconfig-if-aggregate:config":{"lag-type":"LACP","min-links":2,"openconfig-interfaces-ext:fallback":true}}

curl -kX GET "https://10.11.203.17/restconf/data/openconfig-interfaces:interfaces/interface=PortChannel200/openconfig-if-aggregate:aggregation" 
{"openconfig-if-aggregate:aggregation":{"config":{"lag-type":"LACP","min-links":2,"openconfig-interfaces-ext:fallback":true},"openconfig-vlan:switched-vlan":{"config":{"access-vlan":100},"state":{"access-vlan":100}},"state":{"lag-type":"LACP","member":["Ethernet20"],"min-links":2,"openconfig-interfaces-ext:fallback":true}}}

curl -kX GET "https://10.11.203.17/restconf/data/openconfig-interfaces:interfaces/**interface=PortChannel200"
{"openconfig-interfaces:interface":[{"config":{"name":"PortChannel200"},"name":"PortChannel200","openconfig-if-aggregate:aggregation":{"config":{"lag-type":"LACP","min-links":2,"openconfig-interfaces-ext:fallback":true},"openconfig-vlan:switched-vlan":{"config":{"access-vlan":100},"state":{"access-vlan":100}},"state":{"lag-type":"LACP","member":["Ethernet20"],"min-links":2,"openconfig-interfaces-ext:fallback":true}},"openconfig-if-ethernet:ethernet":{"state":{"counters":{"in-oversize-frames":"0","openconfig-if-ethernet-ext:in-distribution":{"in-frames-128-255-octets":"0"},"openconfig-interfaces-ext:out-oversize-frames":"0"}}},"state":{"admin-status":"DOWN","counters":{"in-broadcast-pkts":"0","in-discards":"0","in-errors":"0","in-multicast-pkts":"0","in-octets":"0","in-pkts":"0","in-unicast-pkts":"0","last-clear":"0","out-broadcast-pkts":"0","out-discards":"0","out-errors":"0","out-multicast-pkts":"0","out-octets":"0","out-pkts":"0","out-unicast-pkts":"0"},"enabled":false,"mtu":9100,"name":"PortChannel200","oper-status":"DOWN"},"subinterfaces":{"subinterface":[{"index":0}]}}]}

 curl -kX GET "https://10.11.203.17/restconf/data/openconfig-interfaces:interfaces" 
.
.
.
{"config":{"name":"PortChannel200"},"name":"PortChannel200","openconfig-if-aggregate:aggregation":{"config":{"lag-type":"LACP","min-links":2,"openconfig-interfaces-ext:fallback":true},"openconfig-vlan:switched-vlan":{"config":{"access-vlan":100},"state":{"access-vlan":100}},"state":{"lag-type":"LACP","member":["Ethernet20"],"min-links":2,"openconfig-interfaces-ext:fallback":true}},"openconfig-if-ethernet:ethernet":{"state":{"counters":{"in-oversize-frames":"0","openconfig-if-ethernet-ext:in-distribution":{"in-frames-128-255-octets":"0"},"openconfig-interfaces-ext:out-oversize-frames":"0"}}},"state":{"admin-status":"DOWN","counters":{"in-broadcast-pkts":"0","in-discards":"0","in-errors":"0","in-multicast-pkts":"0","in-octets":"0","in-pkts":"0","in-unicast-pkts":"0","last-clear":"0","out-broadcast-pkts":"0","out-discards":"0","out-errors":"0","out-multicast-pkts":"0","out-octets":"0","out-pkts":"0","out-unicast-pkts":"0"},"enabled":false,"mtu":9100,"name":"PortChannel200","oper-status":"DOWN"},"subinterfaces":{"subinterface":[{"index":0}]}},{"config":{"name":"Vlan100"},"name":"Vlan100","state":{"admin-status":"UP","enabled":true,"mtu":9100,"name":"Vlan100"},"subinterfaces":{"subinterface":[{"index":0}]}},{"config":{"description":"Management0","enabled":true,"mtu":1500,"name":"eth0"},"name":"eth0","openconfig-if-ethernet:ethernet":{"state":{"auto-negotiate":true,"port-speed":"openconfig-if-ethernet:SPEED_1GB"}},"state":{"admin-status":"UP","counters":{"in-discards":"0","in-errors":"0","in-multicast-pkts":"175623","in-octets":"1056137870","in-pkts":"1123049","out-discards":"0","out-errors":"0","out-octets":"421963229","out-pkts":"559106"},"description":"Management0","enabled":true,"mtu":1500,"name":"eth0","oper-status":"UP"},"subinterfaces":{"subinterface":[{"index":0}]}}]}}

**Pull request closed without merging**